### DISCUSSION TITLE
2.33 and a recent git clone fails to compile

### DISCUSSION THREAD

**Discussion by ryanpg123** [ElectroTechnique/TSynth-Teensy4.1#131] [T4.1 code development]
/home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/effect_ensemble.cpp: In member function 'int16_t AudioEffectEnsemble::interpBuffer(float)':
/home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/effect_ensemble.cpp:267:16: error: ordered comparison of pointer with integer zero ('char* (*)(const char*, int)' and 'int')
  267 |   else if(index<0) index2+=ENSEMBLE_BUFFER_SIZE;
      |           ~~~~~^~
In file included from /home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/Audio.h:67,
                 from /home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/TSynth.ino:48:
/home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/synth_dc.h: In member function 'void AudioSynthWaveformDcTS::amplitude(float, float)':
/home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/synth_dc.h:118:17: error: 'void AudioSynthWaveformDcTS::amplitude(float, float)' causes a section type conflict with 'int compare(const void*, const void*)'
  118 |   FLASHMEM void amplitude(float n, float milliseconds) {
      |                 ^~~~~~~~~
In file included from /home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/TSynth.ino:60:
/home/ryanpg/src/TSynth-Teensy4.1-2.33/TSynth/PatchMgr.h:74:14: note: 'int compare(const void*, const void*)' was declared here
   74 | FLASHMEM int compare(const void *a, const void *b) {
      |              ^~~~~~~

exit status 1

Compilation error: ordered comparison of pointer with integer zero ('char* (*)(const char*, int)' and 'int')

**Comment by ryanpg123** [comment]
chatGPT and I fixed this by:

in synth_dc.h

  // slowly transition to the new DC level
  void amplitude(float n, float milliseconds) {
    if (milliseconds <= 0.0f) {
      amplitude(n);
      return;
    }

and in effect_ensemble.cpp

  // // check if next sample index needs to be wrapped.
  // index2=index1+1;
  // if(index2>(ENSEMBLE_BUFFER_SIZE-1)) index2-=ENSEMBLE_BUFFER_SIZE;
  // else if(index<0) index2+=ENSEMBLE_BUFFER_SIZE;

index2 = index1 + 1;
if (index2 > (ENSEMBLE_BUFFER_SIZE - 1)) {
    index2 -= ENSEMBLE_BUFFER_SIZE;
} else if (index2 < 0) {
    index2 += ENSEMBLE_BUFFER_SIZE;
}

**Comment by ryanpg123** [comment]
I still get the following errors and warnings, but it compiles and uploads:

/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp: In member function 'void Threads::idle()':
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:642:7: warning: this 'if' clause does not guard... [-Wmisleading-indentation]
  642 |       if (! threadp[i]) continue;
      |       ^~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:643:25: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  643 |                         if (threadp[i]->sleep_time_till_end_tick > threadp[j]->sleep_time_till_end_tick) {
      |                         ^~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:675:7: warning: this 'if' clause does not guard... [-Wmisleading-indentation]
  675 |       if (! threadp[i]) continue;
      |       ^~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:676:25: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  676 |                         if (needs_run[i]) {
      |                         ^~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp: At global scope:
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:332:13: warning: 'void context_pit_empty()' defined but not used [-Wunused-function]
  332 | static void context_pit_empty() {}
      |             ^~~~~~~~~~~~~~~~~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp: In constructor 'Threads::Threads()':
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:223:42: warning: array subscript -2560 is outside array bounds of 'long unsigned int [1]' [-Warray-bounds]
  223 |   threadp[0]->stack = (uint8_t*)&_estack - DEFAULT_STACK0_SIZE;
      |                       ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:65:22: note: while referencing '_estack'
   65 | extern unsigned long _estack;   // the main thread 0 stack
      |                      ^~~~~~~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:443:6: warning: array subscript -2560 is outside array bounds of 'long unsigned int [1]' [-Warray-bounds]
  443 |   *m = thread_marker;
      |   ~~~^~~~~~~~~~~~~~~
/home/ryanpg/Arduino/libraries/TeensyThreads/TeensyThreads.cpp:65:22: note: while referencing '_estack'
   65 | extern unsigned long _estack;   // the main thread 0 stack
      |                      ^~~~~~~

**Comment by Electrotechnique** [comment]
This is fixed in V2.35 - thanks.
